src/pybind/mgr/volumes/fs/async_cloner.py

```diff
@@ -421,10 +423,40 @@ def cancel_job(self, volname, job):
                         with open_subvol(self.fs_client.mgr, fs_handle, self.vc.volspec, group, clonename, SubvolumeOpType.CLONE_CANCEL) as clone_subvolume:
                             if not self._cancel_job(volname, (track_idx, clone_subvolume.base_path)):
                                 raise VolumeException(-errno.EINVAL, "cannot cancel -- clone finished (check clone status)")
+                            clone_subvolume.add_clone_cancelled(track_idx, clonename)
         except (IndexException, MetadataMgrException) as e:
             log.error("error cancelling clone {0}: ({1})".format(job, e))
             raise VolumeException(-errno.EINVAL, "error canceling clone")
 
+    def bulk_cancel_pending_clone(self, volname, subvolname, snapname, groupname):
+        try:
+            with open_volume(self.fs_client, volname) as fs_handle:
+                with open_group(fs_handle, self.vc.volspec, groupname) as group:
+                    with open_subvol(self.fs_client.mgr, fs_handle, self.vc.volspec, group, subvolname, SubvolumeOpType.CLONE_CANCEL) as subvolume:
+                        pending_clones = subvolume.get_pending_clones(snapname)
```

Comment:
You're getting the ```pending_clones``` list, but then you never use it! You're instead iterating through *every* clone (which includes the completed ones, right?) in the next for loop. That seems wrong?